### DISCUSSION TITLE
Login Password Fails using Server install script

### DISCUSSION THREAD

**Discussion by radiorob** [openbroadcaster/observer#30] [Q&A]
In a rare non-reproducible condition Installer might not create password correctly

First we need the value of OB_HASH_SALT from the config file.

Append that value to the password you want to use. If the salt is aaaaaaaaaa and the password is password, the string needed is aaaaaaaaaapassword

from there, get the sha value like this:

echo -n "aaaaaaaaaapassword" | openssl sha1 | cut -f 2 -d " "

copy the string that this puts out.

the log into mysql:

mysql -u root -p

select database:

use openbroadcaster

UPDATE users SET password='d8656d3f3823bce79e3a29253cf1106f6cfeee7c' WHERE username='admin';

Now, assuming that the salt is set as aaaaaaaaaa in the config file, the password password will work.

Note: When using special characters in the password, they need to use single quotes instead of double quotes when they echo the string

**Comment by PurpleNinja225** [comment]
Confirmed had to follow these instructions to get logged in on a new install using the bash command.  on ubuntu server 18.04.4

**Reply by wdsmaduranga** [reply]
how to install on vps

**Comment by PurpleNinja225** [comment]
Noticed that on a clean install if you view the users table it shows this 
![Screenshot from 2020-07-27 00-11-00](https://user-images.githubusercontent.com/5977403/88519032-ee1e4e00-cf9d-11ea-8a9b-e99386411453.png)

**Comment by radiorob** [comment]
Based on value of the password field seems like a race condition where the file password_hash.php is not where it needs to be by the time the script needs it.  maybe a copy is still in progress at the time?  maybe a test if file exists before making password loop would fix it?

**Comment by Phadungphon-Phatthana** [comment]
how to fix

**Comment by jengajenga** [comment]
4 years later, this worked for me. Thanks.
